travis/build_total.py

```diff
@@ -0,0 +1,180 @@
+        m = re.search(res_word, st, re.MULTILINE)
+        if m:
+            self.res_pr_num = int(m.group(1))
+        m = re.search(ert_word, st, re.MULTILINE)
+        if m:
+            self.ert_pr_num = int(m.group(1))
+   
+    def check_pr_num_consistency(self):
+        if (self.repository == "ecl"):
+            pr_num = self.ecl_pr_num
+        if (self.repository == "res"):
+            pr_num = self.res_pr_num
+        if (self.repository == "ert"):
+            pr_num = self.ert_pr_num
+        if (pr_num >= 0):
+            if (pr_num != self.pr_number):
+                sys.exit("Error: The line rep=%d does not match pull request %d" % (pr_num, self.pr_number))
+                
+        else:
+            if (self.repository == "ecl"):
+                self.ecl_pr_num = self.pr_number
+            if (self.repository == "res"):
+                self.res_pr_num = self.pr_number
+            if (self.repository == "ert"):
+                self.ert_pr_num = self.pr_number
+
+    def assert_open_pr_status(self, rep_name, pr_num):
+        if (pr_num >= 0):
+            url = "https://api.github.com/repos/Statoil/%s/pulls/%d" % (rep_name, pr_num)
+            if "GITHUB_API_TOKEN" in os.environ:
+                github_api_token = os.getenv("GITHUB_API_TOKEN")
+                response = requests.get( url , {"access_token" : github_api_token})
+            else:
+                response = requests.get( url )
+            content = json.loads( response.content )
+            state = content["state"]
+            assert(state == "open")
+            
+    def access_pr(self): 
+        pr_number_string = os.getenv("TRAVIS_PULL_REQUEST")
+        self.pr_number = int(pr_number_string)
+        url = "https://api.github.com/repos/Statoil/%s/pulls/%d" % (self.rep_name, self.pr_number)
+        print "Accessing: %s" % url
+
+        if "GITHUB_API_TOKEN" in os.environ: 
+            github_api_token = os.getenv("GITHUB_API_TOKEN")
+            response = requests.get( url , {"access_token" : github_api_token})
+        else:
+            response = requests.get( url )
+    
+        content = json.loads( response.content )
+        self.pr_description = content["body"]
+        print "PULL REQUEST: %d\n%s" % (self.pr_number, self.pr_description)
+        self.parse_pr_description();
+        self.check_pr_num_consistency();
+        self.assert_open_pr_status('libecl', self.ecl_pr_num)
+        self.assert_open_pr_status('libres', self.res_pr_num)
+        self.assert_open_pr_status('ert'   , self.ert_pr_num)
+
+    def clone_fetch_merge(self, basedir):      
+        self.clone_merge_repository('libecl', self.ecl_pr_num, basedir)
+        self.clone_merge_repository('libres', self.res_pr_num, basedir)
+        self.clone_merge_repository('ert'   , self.ert_pr_num, basedir)
+
+    def clone_merge_repository(self, rep_name, pr_num, basedir):
+        if (self.rep_name != rep_name):
+            os.chdir(basedir)
```

Comment:
This is dangerous; unless the name of the method is something like `chdir()` one method should not leave a net change in current working directory - i.e. within one function all `chdir( ... )` calls should come in *pairs* like:

```python
cwd0 = os.getcwd( )
os.chdir( path )
....
os.chdir( cwd0 )
```